bot/exts/utils/jams.py

```diff
@@ -20,124 +22,155 @@ class CodeJams(commands.Cog):
+                await self.create_team_channel(ctx.guild, team_name, members, team_leaders)
 
-    async def get_category(self, guild: Guild) -> CategoryChannel:
+            await self.create_team_leader_channel(ctx.guild, team_leaders)
+            await ctx.send(f"{Emojis.check_mark} Created Code Jam with {len(teams)} teams.")
+
+    async def get_category(self, guild: discord.Guild) -> discord.CategoryChannel:
         """
         Return a code jam category.
 
         If all categories are full or none exist, create a new category.
         """
         for category in guild.categories:
             # Need 2 available spaces: one for the text channel and one for voice.
-            if category.name == CATEGORY_NAME and MAX_CHANNELS - len(category.channels) >= 2:
+            if category.name == CATEGORY_NAME and len(category.channels) < MAX_CHANNELS:
                 return category
 
         return await self.create_category(guild)
 
-    @staticmethod
-    async def create_category(guild: Guild) -> CategoryChannel:
+    async def create_category(self, guild: discord.Guild) -> discord.CategoryChannel:
         """Create a new code jam category and return it."""
         log.info("Creating a new code jam category.")
 
         category_overwrites = {
-            guild.default_role: PermissionOverwrite(read_messages=False),
-            guild.me: PermissionOverwrite(read_messages=True)
+            guild.default_role: discord.PermissionOverwrite(read_messages=False),
+            guild.me: discord.PermissionOverwrite(read_messages=True)
         }
 
-        return await guild.create_category_channel(
+        category = await guild.create_category_channel(
             CATEGORY_NAME,
             overwrites=category_overwrites,
             reason="It's code jam time!"
         )
 
+        await self.send_status_update(
+            guild, f"Created a new category with the ID {category.id} for this Code Jam's team channels."
+        )
+
+        return category
+
     @staticmethod
-    def get_overwrites(members: t.List[Member], guild: Guild) -> t.Dict[t.Union[Member, Role], PermissionOverwrite]:
+    def get_overwrites(
+        members: list[tuple[discord.Member, bool]],
+        guild: discord.Guild,
+    ) -> dict[t.Union[discord.Member, discord.Role], discord.PermissionOverwrite]:
         """Get code jam team channels permission overwrites."""
-        # First member is always the team leader
         team_channel_overwrites = {
-            members[0]: PermissionOverwrite(
-                manage_messages=True,
-                read_messages=True,
-                manage_webhooks=True,
-                connect=True
-            ),
-            guild.default_role: PermissionOverwrite(read_messages=False, connect=False),
+            guild.default_role: discord.PermissionOverwrite(read_messages=False),
+            guild.get_role(Roles.moderators): discord.PermissionOverwrite(read_messages=True),
```

Comment:
We're not planning on giving mods access.

Reply:
I guess this changed from the original issue, fixed in https://github.com/python-discord/bot/pull/1668/commits/37f8637955350c6c5b437d66807f42d8f26adfcd.